woocommerce/woocommerce
# GH Action for generating reports in CI

### Changes proposed in this Pull Request:

Closes 225-gh-woocommerce/woocommerce-quality.

> NOTE:
>
> This PR is based off of [Allure Framework integration to generate HTML reports locally](https://github.com/woocommerce/woocommerce/pull/32091). Please merge that PR first before this.

This PR adds a job in the `Build zip for PR` workflow to upload the API test report as an artifact. The artifact contains the `allure-results` and `allure-report` folders that Allure generated by the `Runs API tests` job.

<img width="1017" alt="Screen Shot 2022-03-29 at 2 28 39 PM" src="https://user-images.githubusercontent.com/4509348/160547928-ab5d5f2c-6350-40f7-914f-efa5d5150c95.png">


### How to test the changes in this Pull Request:

1. Create a new branch that's based off from this branch.
2. Create a draft PR of your branch.
3. Wait for the checks to finish.
4. Navigate to the `Build zip for PR` run for your draft PR, and verify that an artifact named `api-test-report---pr-NNNNN` was created. Make sure the PR number in the artifact name is correct. 

### Other information:




### Changelog entry

> Added: CI job to upload API test reports on PR events.

### FOR PR REVIEWER ONLY:
